src/client/datascience/editor-integration/codewatcher.ts

```diff
@@ -163,11 +165,7 @@ export class CodeWatcher implements ICodeWatcher {
 
                 // Note: We do a get or create active before all addCode commands to make sure that we either have a history up already
                 // or if we do not we need to start it up as these commands are all expected to start a new history if needed
-                const success = await this.addCode(code, this.document.uri, range.start.line);
-                if (!success) {
-                    await this.addErrorMessage(this.document.uri, leftCount);
-                    break;
-                }
+                this.queueAddCode(code, this.document.uri, range.start.line, leftCount);
```

Comment:
Previous code here just loops and awaits. So if you did run cells [A, B, C] and then run cells [A, B, C] you would get [A, A, B, B, C, C] in the output, not the expected [A, B, C, A, B, C]. Pretty easy to cause other ordering issues as well with the other cell commands.